APCAppCore/APCAppCore/Library/FeedParser/APCFeedParser.m

```diff
@@ -0,0 +1,259 @@
+    
+    if (!success) {
+        
+        if (weakSelf.completionBlock) {
+            weakSelf.completionBlock(nil, weakSelf.parser.parserError);
+        }
+    }
+}
+
+#pragma mark - NSXMLParserDelegate methods
+
+- (void)parser:(NSXMLParser *)__unused parser didStartElement:(NSString *)elementName namespaceURI:(NSString *)__unused namespaceURI qualifiedName:(NSString *)__unused qName attributes:(NSDictionary *)attributeDict
+{
+    self.currentElement = elementName;
+    self.attributeDict = attributeDict;
+    
+    if ([elementName isEqualToString:@"item"] || [elementName isEqualToString:@"entry"]) {
+        self.feedItem = [APCFeedItem new];
+    }
+    
+    self.parsedString = [NSMutableString new];
+}
+
+- (void)parser:(NSXMLParser *)__unused parser foundCharacters:(NSString *)string
+{
+    [self.parsedString appendString:string];
+}
+
+- (void)parser:(NSXMLParser *)__unused parser didEndElement:(NSString *)elementName namespaceURI:(NSString *)__unused namespaceURI qualifiedName:(NSString *)__unused qName
+{
+    if ([elementName isEqualToString:@"item"] || [elementName isEqualToString:@"entry"]) {
+        [self.results addObject:self.feedItem];
+    }
+    
+    if (self.feedItem != nil && self.parsedString != nil) {
+        if ([elementName isEqualToString:@"title"]) {
+            self.feedItem.title = self.parsedString;
+        } else if ([elementName isEqualToString:@"description"]) {
+            self.feedItem.itemDescription = self.parsedString;
+        } else if ([elementName isEqualToString:@"content:encoded"] || [elementName isEqualToString:@"content"]) {
+            self.feedItem.content = self.parsedString;
+        } else if ([elementName isEqualToString:@"link"]) {
+            self.feedItem.link = self.parsedString;
+        } else if ([elementName isEqualToString:@"pubDate"]) {
+            _dateFormatter.dateFormat = @"EEE, dd MMM yyyy HH:mm:ss Z";
+            self.feedItem.pubDate = [self.dateFormatter dateFromString:self.parsedString];
+        } else if ([elementName isEqualToString:@"dc:creator"]) {
+            self.feedItem.author = self.parsedString;
+        } else if ([elementName isEqualToString:@"guid"]) {
+            self.feedItem.guid = self.parsedString;
+        }
+        
+        // sometimes the URL is inside enclosure element, not in link. Reference: http://www.w3schools.com/rss/rss_tag_enclosure.asp
+        if ([elementName isEqualToString:@"enclosure"] && self.attributeDict != nil) {
+            NSString *url = [self.attributeDict objectForKey:@"url"];
+            if(url) {
+                self.feedItem.link = url;
+            }
+        }
+    }
+}
+
+- (void)parserDidEndDocument:(NSXMLParser *)__unused parser
+{
+    __weak typeof(self) weakSelf = self;
+    
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if (weakSelf.completionBlock) {
+            weakSelf.completionBlock([NSArray arrayWithArray:self.results], nil);
+        }
```

Comment:
Not just that. Also, you should NOT reference weakSelf multiple times in the same block, because you might get different answers back. Instead, do `__strong typeof(self) strongSelf = weakSelf` inside the block and then use strongSelf as much as you want.